enhancements/ingress/set-delete-http-headers.md

```diff
@@ -0,0 +1,1120 @@
+| `proxy`                    | None                                                | None                                                              | None                                                  | None                                                                        | None                                                 |
+| `host`                     | None                                                | None                                                              | Yes                                                   | None                                                                        | None                                                 |
+| `strict-transport-security` | `haproxy.router.openshift.io/hsts_header`           | None                                                              | None                                                  | Yes                                                                         | Yes                                                  |                                                     
+| `x-forwarded-for`          | `haproxy.router.openshift.io/set-forwarded-headers` | Yes                                                               | Yes                                                   | Yes(using API spec.httpHeaders.forwardedHeaderPolicy) except delete action) | Yes(using route annotation)                          |
+| `x-ssl`                    | None                                                | Yes                                                               | Yes                                                   | None                                                                        | None                                                 |                    
+
+
+### Drawbacks
+
+Please refer to the `Risks and Mitigations` section to understand which headers are not allowed to be set/delete using this EP.
+
+## Design Details
+
+### Open Questions
+
+#### Will this enhancement cover setting or deleting request headers in addition to response headers?
+  Answer: Yes, we will be covering setting or deleting of request headers.
+  This is required for the following use-case.  
+  
+  As a cluster admin or route owner, I want to enable mutual TLS but replace the "x-ssl-client-der" request header
+  that OpenShift router adds with an "x-forwarded-client-cert" request header that my application checks for, 
+  so that I can migrate my applications that rely on mutual TLS and use the latter header name. 
+
+  The following example replaces "x-ssl-client-der" with "x-forwarded-client-cert":
+```yaml
+spec:
+  httpHeaders:
+    actions:
+      response:
+      - name: X-SSL-Client-Cert
+        action:
+        type: Set
+        set:
+          value: "%{+Q}[ssl_c_der,base64]"
+      - name:   Content-Language
+        action:
+        type: Delete
```

Comment:
This example yaml is wrong.  The text above says that the _request_ headers should be modified by adding "x-forwarded-client-cert" and removing "x-ssl-client-der", so the configuration should be something more like this:
```suggestion
      request: 
      - name: x-forwarded-client-cert
        action: Set
        set:
          value: "%{+Q}[ssl_c_der,base64]"
      - name: x-ssl-client-der
        action: Delete
```

Reply:
done